KFSPTS-28468 update variable names in Jaggaer Supplier DTO objects

jhulslander opened the pull request:
XJC created these pojos with some confusing variable names and structures.  This PR attempts to fix some of the issues.

ajd299 commented on src/test/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/JaggaerExampleTest.java:
```diff
@@ -430,14 +430,14 @@ private DiversityClassificationList buildDiversityClassificationList() {
         AdditionalDataList dataList = new AdditionalDataList();
         
-        AdditionalData dataum = new AdditionalData();
+        AdditionalDataItem dataum = new AdditionalDataItem();
```
what does the variable name "dataum" mean?

jhulslander replied:
looks like this was probably a typo on my part, I will fix it.  This should be datum, which is the singular form data

dp65 replied:
I know that this has already been approved, merged, and probably promoted to production, but I just saw this today, and I want to jump in. 

Using "datum" is fraught with opinion. As far as I've noticed in the past 20 years, I've never heard anyone say, "those data were written to the log file." In most situations, "data" is commonly used as a mass noun, like "information", to refer to a collection of more than one datum, and a collection is singular. Furthermore the number of times I have actually heard someone use the word "datum" has declined to almost zero.

In the case of this code, the class is called "AdditionalDataItem", so for consistency, the least confusing variable name would be "dataItem", not "datum", like what was used on line 431 with "AdditionalDataList dataList = ...".

Just sticking my nose in where it doesn't belong...

ajd299 reviewed (changes requested):
Code looks good, please see my comments/questions

ajd299 reviewed (changes requested):
Code looks good, please see my comments/questions

jhulslander commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/AccountsPayable.java:
```diff
@@ -38,7 +38,7 @@ public class AccountsPayable {
     @XmlElement(name = "Active")
     private Active active;
     @XmlElement(name = "AssociatedAddress")
```
the element name is the XML tag value, we can't change that as it is determined by Jaggaer

jhulslander commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/AdditionalDataItem.java:
```diff
@@ -15,9 +15,9 @@
 @XmlAccessorType(XmlAccessType.FIELD)
-@XmlType(name = "", propOrder = { "content" })
+@XmlType(name = "", propOrder = { "contents" })
 @XmlRootElement(name = "AdditionalData")
```
the prop order is related to the xml tags, it control the valid tags and the order the must be present. this is controlled by jaggaer

jhulslander commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/AdditionalDataList.java:
```diff
@@ -30,11 +30,11 @@ public void setIsChanged(String value) {
+            additionalDataItems = new ArrayList<AdditionalDataItem>();
         }
-        return this.additionalData;
+        return this.additionalDataItems;
```
I am not sure we have a "Standard" convention, but if we use tools to create getters and setters for us, they include "this" and I think that is fine.  Perhaps this could be a book club topic.  I would rather defer removing "this" from the getters and setters if it turns out we need to.

jhulslander replied:
I will see about getting that clean up here

jhulslander commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/AttachmentList.java:
```diff
@@ -14,15 +14,15 @@
+@XmlType(name = "", propOrder = { "attachments" })
 @XmlRootElement(name = "Attachments")
-public class Attachments {
+public class AttachmentList implements CustomElementDetail {
```
You can see it used in the variable customElementDetails in the  CustomElement class.  The interface is implemented on AttachmentList and CustomElementValueList.  It is just an easy way to control what object types can be added to customElementDetails.  
This model is used a number of times in these Jaggaer pojos

jhulslander commented:
I think I have answered the questions above, and made some minor changes.  This is ready for review again

cah292 commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/SecondaryNaicsList.java:
```diff
@@ -12,15 +12,15 @@
     private String isChanged;
     @XmlElement(name = "SecondaryNaics")
-    private List<SecondaryNaics> secondaryNaics;
+    private List<SecondaryNaicsItem> secondaryNaicItems;
```
Many of your other objects are still keeping the "s" in "naics". Should this property be named "secondaryNaicsItems" instead for consistency?

jhulslander replied:
ah, yes, I will fix

cah292 commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/TaxDocument.java:
```diff
@@ -28,12 +28,12 @@ public void setIsChanged(String isChanged) {
 
-    public void setAttachments(Attachments attachments) {
-        this.attachments = attachments;
+    public void setAttachmentList(AttachmentList attachments) {
```
The method argument's name should be updated to match the new property name.

cah292 commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/TaxInformationList.java:
```diff
@@ -12,15 +12,15 @@
     private String isChanged;
     @XmlElement(name = "TaxInformation")
-    private List<TaxInformation> taxInformation;
+    private List<TaxInformation> taxInformations;
```
Another list property that might work out better with a name suffix of "Items" or "Entries" or "Details".

jhulslander replied:
switched to taxInformationDetails

cah292 commented on src/test/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/SupplierSyncMessageTest.java:
```diff
@@ -338,16 +338,16 @@ private TaxInformationList buildTaxInformationList() {
-    private Attachments buildAttachments(String attachmentName) {
-        Attachments attachments = new Attachments();
+    private AttachmentList buildAttachments(String attachmentName) {
+        AttachmentList attachments = new AttachmentList();
```
Should this variable be renamed to "attachmentList" for consistency?

jhulslander replied:
ok

cah292 reviewed (changes requested):
I had a few suggestions for some naming improvements, but overall the changes look good.

jhulslander commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/AdditionalDataItem.java:
```diff
@@ -27,7 +26,7 @@ public class AdditionalData {
     @XmlMixed
     @XmlAnyElement
-    private List<Object> content;
+    private List<String> contents;
```
I will switch it back to Object

jhulslander commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/AccountsPayableList.java:
```diff
@@ -12,15 +12,15 @@
     private String isChanged;
     @XmlElement(name = "AccountsPayable")
-    private List<AccountsPayable> accountsPayable;
+    private List<AccountsPayable> accountsPayables;
```
I will switch to accountsPayableDetails

jhulslander commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/InsuranceInformationList.java:
```diff
@@ -12,15 +12,15 @@
     private String isChanged;
     @XmlElement(name = "InsuranceInformation")
-    private List<InsuranceInformation> insuranceInformation;
+    private List<InsuranceInformation> insuranceInformations;
```
I will switch to insuranceInformationDetails

jhulslander commented:
this is ready for review again
I have removed the extra "THIS"  calls from the getters and I went back through the POJOs and tried to use the naming conventions discussed in this PR.  I have missed some, please let me know if you notice things awry

cah292 reviewed (changes requested):
I noticed some items from the latest revisions that should be cleaned up; see my comments for details. Also, in my prior review I only meant for some of the list properties to be updated with an extra suffix, depending on their wording; I didn't mean that such a change should be applied to every list property. Sorry if that wasn't clear.

jhulslander commented:
This is ready for review again .... I ended up reverting changes made earlier today and then just adding in the changes noted on the PR.  I also removed "THIS" usage in the collection getters.  Lastly I updated the interface name convention you noted earlier

jhulslander commented:
retest this please

cah292 commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/Discount.java:
```diff
@@ -44,9 +44,9 @@ public void setUnit(String unit) {
-            discountItems = new ArrayList<DiscountItemInterface>();
+            discountItems = new ArrayList<DiscountItem>();
         }
         return this.discountItems;
```
You could remove the `this` usage here.

jhulslander replied:
ok

cah292 commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/DiscountAmount.java:
```diff
@@ -12,7 +12,7 @@
 @XmlType(name = "", propOrder = { "value" })
 @XmlRootElement(name = "DiscountAmount")
-public class DiscountAmount implements DiscountItemInterface{
+public class DiscountAmount implements DiscountItem{
```
Missing space before curly brace.

jhulslander replied:
ok

cah292 commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/DiscountPercent.java:
```diff
@@ -12,7 +12,7 @@
 @XmlType(name = "", propOrder = { "value" })
 @XmlRootElement(name = "DiscountPercent")
-public class DiscountPercent implements DiscountItemInterface{
+public class DiscountPercent implements DiscountItem{
```
Missing space before curly brace.

jhulslander replied:
ok

cah292 commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/IsoCurrencyCode.java:
```diff
@@ -12,7 +12,7 @@
 @XmlType(name = "", propOrder = { "value" })
 @XmlRootElement(name = "IsoCurrencyCode")
-public class IsoCurrencyCode implements DiscountItemInterface{
+public class IsoCurrencyCode implements DiscountItem{
```
Missing space before curly brace.

jhulslander replied:
ok

cah292 commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/SupplierRequestMessage.java:
```diff
@@ -9,18 +9,18 @@
+@XmlType(name = "", propOrder = { "suppliers" })
 @XmlRootElement(name = "SupplierRequestMessage")
-public class SupplierRequestMessage implements SupplierRequestMessageItemInterface{
+public class SupplierRequestMessage implements SupplierRequestMessageItem{
```
Missing space before curly brace.

jhulslander replied:
ok

cah292 reviewed (changes requested):
I noticed a few minor items that would be good to clean up, but other than that the changes look good.

cah292 commented on src/main/java/edu/cornell/kfs/module/purap/jaggaer/supplier/xml/CustomElementList.java:
```diff
@@ -12,15 +12,15 @@
     private String isChanged;
     @XmlElement(name = "CustomElement")
-    private List<CustomElement> customElement;
+    private List<CustomElement> customElementDetails;
```
Now that the code no longer has a "customElementItems" property, I'm fine with the usage of the "customElementDetails" name.

jhulslander commented:
this is ready for review again

jhulslander commented:
@cah292  I was able to dismiss Andrew's review as an admin

cah292 commented:
Thanks! I'll merge it now.
